### PR TITLE
Asynchronous storage of data in ccd

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/idam/services/UserService.java [hunk 1]
```diff
@@ -13,40 +18,81 @@
...
 
-    private String getBasicAuthHeader(String username, String password) {
-        String auth = username + ":" + password;
-        return "Basic " + new String(Base64.getEncoder().encode(auth.getBytes()));
+    public String getBasicAuthHeader(String username, String password) {
+        String authorisation = username + ":" + password;
+        return BASIC + Base64.getEncoder().encodeToString(authorisation.getBytes());
     }
 
+    public String getIdamOauth2Token(String username, String password) {
```
Comment (timja) [inline ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/idam/services/UserService.java]: if you're making changes here, you should look at using this instead:
https://github.com/hmcts/idam-java-client

Comment (timja) [reply ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/idam/services/UserService.java]: https://github.com/hmcts/idam-java-client/blob/master/src/main/java/uk/gov/hmcts/reform/idam/client/IdamClient.java#L32

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/CaseMapper.java [hunk 2]
```diff
@@ -79,6 +90,9 @@ public CCDCase to(Claim claim) {
                 .build())
         );
 
+        claim.getClaimantResponse()
+            .ifPresent(claimantResponse -> builder.claimantResponse(claimantResponseMapper.to(claimantResponse)));
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/CaseMapper.java]: Generally preferred to make lambdas atomic and chained.
e.g. if you agree: `claim.getClaimantResponse().map(claimandResponseMapper::to).ifPresent(builder::claimantResponse)`

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/claimantresponse/ResponseAcceptationMapper.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    public CCDResponseAcceptation to(ResponseAcceptation responseAcceptation) {
+        CCDResponseAcceptation.CCDResponseAcceptationBuilder builder = CCDResponseAcceptation.builder();
+        builder.amountPaid(responseAcceptation.getAmountPaid())
+            .formaliseOption(CCDFormaliseOption.valueOf(responseAcceptation.getFormaliseOption().name()));
+
+        responseAcceptation.getClaimantPaymentIntention()
+            .ifPresent(paymentIntention -> builder.claimantPaymentIntention(paymentIntentionMapper.to(paymentIntention)));
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/claimantresponse/ResponseAcceptationMapper.java]: [as above](https://github.com/hmcts/cmc-claim-store/pull/515/files#r227323680)

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/claimantresponse/ResponseAcceptationMapper.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+        responseAcceptation.getCourtDetermination()
+            .ifPresent(courtDetermination -> builder.courtDetermination(courtDeterminationMapper.to(courtDetermination)));
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/claimantresponse/ResponseAcceptationMapper.java]: [as above](https://github.com/hmcts/cmc-claim-store/pull/515/files#r227323680)

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/claimantresponse/ResponseRejectionMapper.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            .build();
+    }
+
+    @Override
+    public ResponseRejection from(CCDResponseRejection ccdResponseRejection) {
+        ResponseRejection.ResponseRejectionBuilder builder = ResponseRejection.builder()
+            .amountPaid(ccdResponseRejection.getAmountPaid())
+            .reason(ccdResponseRejection.getReason());
+
+        if(ccdResponseRejection.getFreeMediationOption() != null){
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/claimantresponse/ResponseRejectionMapper.java]: Missing some whitespace formatting

**Pull request closed without merging**